packages/x-data-grid/src/internals/demo/DemoContainer.tsx

```diff
@@ -0,0 +1,42 @@
+import * as React from 'react';
+import { GridRootPropsContext } from '../../context/GridRootPropsContext';
+import { GridContextProvider } from '../../context/GridContextProvider';
+import { useDataGridProps } from '../../DataGrid/useDataGridProps';
+import { useDataGridComponent } from '../../DataGrid/useDataGridComponent';
+import { useGridAriaAttributes } from '../../hooks/utils/useGridAriaAttributes';
+import { useGridRowAriaAttributes } from '../../hooks/features/rows/useGridRowAriaAttributes';
+import materialSlots from '../../material';
+
+interface DemoContainerProps {
+  children: React.ReactNode;
+}
+
+const configuration = {
+  hooks: {
+    useGridAriaAttributes,
+    useGridRowAriaAttributes,
+  },
+};
+
+const contextValue = { slots: materialSlots };
+
+/**
+ * WARNING: This is an internal component used in documentation to provide the required context for demos.
+ * Please do not use it in your application.
+ */
+export function DemoContainer(props: DemoContainerProps) {
```

Comment:
This component allows us to render Data Grid subcomponents in an isolated demo, i.e. we can render the grid toolbar without the rest of the data grid.